hackathon_site/hardware/test_api.py

```diff
@@ -188,9 +188,606 @@ def test_hardware_get_success(self):
+        self._login()
+        profile = self._make_event_profile()
+        hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=10,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+        hardware.categories.add(self.category_limit_10.pk)
+
+        request_data = {"hardware": [{"id": hardware.id, "quantity": 2}]}
+        response = self.client.post(self.view, request_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        pass
+
+    def test_invalid_input_hardware_limit_past_orders(self):
+        self._login()
+        profile = self._make_event_profile()
+        hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=10,
+            max_per_team=4,
+            picture="/picture/location",
+        )
+        hardware.categories.add(self.category_limit_10.pk)
+
+        submitted_order = Order.objects.create(
+            team=self.user.profile.team, status="Submitted"
+        )
+        submitted_order_item = OrderItem.objects.create(
+            order=submitted_order, hardware=hardware
+        )
+
+        ready_order = Order.objects.create(
+            team=self.user.profile.team, status="Ready for Pickup"
+        )
+        ready_order_item = OrderItem.objects.create(
+            order=ready_order, hardware=hardware
+        )
+
+        picked_up_order = Order.objects.create(
+            team=self.user.profile.team, status="Picked Up"
+        )
+        picked_up_order_item = OrderItem.objects.create(
+            order=picked_up_order, hardware=hardware
+        )
+
+        request_data = {"hardware": [{"id": hardware.id, "quantity": 2}]}
+        response = self.client.post(self.view, request_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        pass
+
+    def test_hardware_limit_returned_orders(self):
+        self._login()
+        profile = self._make_event_profile()
+        hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=10,
+            max_per_team=4,
+            picture="/picture/location",
+        )
+        hardware.categories.add(self.category_limit_10.pk)
+
+        order = Order.objects.create(team=self.user.profile.team, status="Picked Up")
+        healthy_order_item = OrderItem.objects.create(order=order, hardware=hardware)
+        healthy_order_item.part_returned_health = "Healthy"
```

Comment:
You can do this inline with the `OrderItem.objects.create(..., part_returned_health="Healthy")`